engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneDelegateTest.m

```diff
@@ -22,6 +26,52 @@ - (void)setUp {
 - (void)tearDown {
 }
 
+- (void)testMoveRootViewControllerWhenWindow {
+  id mockApplication = OCMClassMock([UIApplication class]);
+  OCMStub([mockApplication sharedApplication]).andReturn(mockApplication);
+
+  id mockAppDelegate = OCMClassMock([FlutterAppDelegate class]);
+  OCMStub([mockApplication delegate]).andReturn(mockAppDelegate);
+
+  id mockWindow = OCMClassMock([UIWindow class]);
+  OCMStub([mockAppDelegate window]).andReturn(mockWindow);
+
+  id mockRootViewController = OCMClassMock([UIViewController class]);
+  OCMStub([mockWindow rootViewController]).andReturn(mockRootViewController);
+
+  id scene = OCMClassMock([UIWindowScene class]);
+  id session = OCMClassMock([UISceneSession class]);
+  id connectionOptions = OCMClassMock([UISceneConnectionOptions class]);
+
+  FlutterSceneDelegate* sceneDelegate = [[FlutterSceneDelegate alloc] init];
+  id mockSceneDelegate = OCMPartialMock(sceneDelegate);
+  OCMStub([mockSceneDelegate moveRootViewControllerFrom:[OCMArg any] to:[OCMArg any]]);
+
+  [mockSceneDelegate scene:scene willConnectToSession:session options:connectionOptions];
+
+  OCMVerify(times(1), [mockSceneDelegate moveRootViewControllerFrom:[OCMArg any] to:[OCMArg any]]);
```

Comment:
I think we have the information to use proper argument matchers here.

Reply:
True! Done